client.go

```diff
@@ -353,11 +353,50 @@ func (c *Client) ListImages(ctx context.Context) ([]Image, error) {
 	return images, nil
 }
 
-// Events returns an event subscription for the provided filters
-func (c *Client) Events(ctx context.Context, filters ...string) (eventsapi.Events_SubscribeClient, error) {
-	return c.EventService().Subscribe(ctx, &eventsapi.SubscribeRequest{
+// Subscribe to events that match one or more of the provided filters.
+//
+// Callers should listen on both the envelope channal and errs channel. If the
+// errs channel returns nil or an error, the subscriber should terminate.
+//
+// To cancel shutdown reciept of events, cancel the provided context. The errs
+// channel will be closed and return a nil error.
+func (c *Client) Subscribe(ctx context.Context, filters ...string) (ch <-chan *eventsapi.Envelope, errs <-chan error) {
```

Comment:
Whats the advantage of two channels instead of one with a struct containing both the error and envelope?

Reply:
The `errs` channel controls shutdown separately. Consuming code is simpler:

```go
select {
case ev := <- ch:
  // handle event
case err := <-errs:
  if err != nil { return err }
}
```